lib/rw-overlay.nix

```diff
@@ -1,28 +1,28 @@
 #needs to be a runtime eval
-{writeShellScriptBin,
+{writeShellScriptBin, fuse-overlayfs,
 mountprefix, callback, base}:
   writeShellScriptBin "mounter" ''
     set -euo pipefail #http://redsymbol.net/articles/unofficial-bash-strict-mode/
     export SHELLOPTS #Note export exports the key not the value, "updates propagate"
     set -x
 
     mount="${mountprefix}"
+    gio trash "$mount" || true
```

Comment:
We need to ensure the directories are empty, otherwise it will fail to mount. Maybe we could create the directories in random subdirectory.